packages/vm/src/modules/file_system_cache.rs

```diff
@@ -62,14 +62,23 @@ impl FileSystemCache {
         }
     }
 
-    /// Loads an artifact from the file system and returns a module (i.e. artifact + store).
-    pub fn load(&self, checksum: &Checksum, store: &Store) -> VmResult<Option<Module>> {
+    /// Loads an artifact from the file system and returns a module (i.e. artifact + store),
+    /// along with the artifact size.
```

Comment:
```suggestion
    /// Loads a serialized module from the file system and returns a module (i.e. artifact + store),
    /// along with the size of the serialized module.
```

The artifacts are an implementation detail, which we should not operate on. I thought we could do that, but Wasmer made clear that the user interface is always Module.